lib/openapi_client_generator.rb

```diff
@@ -90,7 +90,12 @@ def option_overrides
         end
       end
       if definition.raw["x-github"]["previews"].any? {|e| e["required"]}
-        options << "opts = ensure_api_media_type(:#{namespace}, options)"
+        # this is a bit janky
+        resource = definition.operation_id.split("/").first
+        namespace_segments = namespace.split("_")
+        preview_type = (resource == namespace_segments.last.pluralize) ? resource : namespace
+
+        options << "opts = ensure_api_media_type(:#{preview_type}, options)"
```

Comment:
How can we help not make it janky 😄

Reply:
I'd love to know 😅It'd be cool if we could read the `preview.rb` file. I should look into that.